dsl_parser/exceptions.py

```diff
@@ -31,6 +31,15 @@ def __init__(self, *args, **kwargs):
         super(UnknownInputError, self).__init__(*args, **kwargs)
 
 
+class InputEvaluationError(Exception):
+    """
+    An error raised when the provided input cannot be evaluated (e.g. when
+    it's missing an attribute that has been required)
+    """
+    def __init__(self, *args, **kwargs):
```

Comment:
You don't actually need to specify this if you're not doing anything with the __init__ yourself- just remove is, put a 'pass' in and it'll fall through to the parent's __init__.